### ISSUE TITLE
.Rotate behaves unexpected in a timeline

### ISSUE THREAD

**Issue by markuswinkler** [u10int/Kinetic#5]
If I modify the example from the docs from 

```
let timeline = Timeline()
timeline.add(Kinetic.to(square, duration: 1, options: [ .X(110) ]).ease(Easing.inOutCubic))
timeline.add(Kinetic.to(square, duration: 1, options: [ .Y(250) ]).ease(Easing.inOutCubic))
timeline.add(Kinetic.to(square, duration: 1, options: [ .Scale(2) ]).ease(Easing.inOutCubic))
timeline.play()
```

to

```
let timeline = Timeline()
timeline.add(Kinetic.to(square, duration: 1, options: [ .Rotate(-1) ]).ease(Easing.inOutCubic))
timeline.add(Kinetic.to(square, duration: 1, options: [ .Rotate(1) ]).ease(Easing.inOutCubic))
timeline.add(Kinetic.to(square, duration: 1, options: [ .Rotate(0) ]).ease(Easing.inOutCubic))
timeline.play()
```

the result is not as expected.

**Comment by u10int** [comment]
@markuswinkler There's always some weirdness when trying to do linear interpolation of transform matrices, so I've refactored how transform properties are handled and things are working considerably better. I've tested with your rotation example and your scale example in https://github.com/u10int/Kinetic/issues/6 as well as a few others and everything is working as expected. 

There's still an issue I need to address with replaying an animation timeline that's already played once and the transforms not always animating. I'll open another issue for this.

You can test my latest changes and fixes from master before I push out 0.9.1 later this week after doing some additional cleanup.

**Comment by markuswinkler** [comment]
Great! Works like a charm.
